src/shared/clock-util.c

```diff
@@ -152,11 +155,22 @@ int clock_apply_epoch(void) {
         } else
                 epoch_usec = timespec_load(&st.st_mtim);
 
-        if (now(CLOCK_REALTIME) >= epoch_usec)
+        now_usec = now(CLOCK_REALTIME);
+        if (now_usec > epoch_usec)
+                diff_usec = usec_sub_unsigned(now_usec, epoch_usec);
+        else
+                diff_usec = usec_sub_unsigned(epoch_usec, now_usec);
+
+        if (now_usec < epoch_usec)
+                ret = 1;
+        else if (diff_usec > (usec_t) CLOCK_DIFF_THRESHOLD * USEC_PER_SEC)
+                ret = 2;
```

Comment:
nope. no arbitrary constants like this. C has enums for a reason. Also, use the retur value for error codes please, see below.

Add a new output param:

```c
typedef ClockChangeDirection {
        CLOCK_CHANGE_NOOP
        CLOCK_CHANGE_FORWARD,
        CLOCK_CHANGE_BACKWARD,
} ClockChangeDirection;
```

and then change `clock_apply_epoch()` to a protype like this:

```c
int clock_apply_epoch(ClockChangeDirection *attempted_change) {
…
```

And then return the attempted direction change in that.